modules/markup/html.go

```diff
@@ -657,6 +658,15 @@ func sha1CurrentPatternProcessor(ctx *postProcessCtx, node *html.Node) {
 	// but that is not always the case.
 	// Although unlikely, deadbeef and 1234567 are valid short forms of SHA1 hash
 	// as used by git and github for linking and thus we have to do similar.
+	// Because of this, we check to make sure that a matched hash is actually
+	// a commit in the repository before making it a link.
+	if ctx.metas["repoPath"] != "" {
```

Comment:
There should be some tests added for this, since it doesn't seem to work for short SHAs when I ran it. I've added some debugging and it looks like IsCommitExist eventually returns an Object Not Found error for a short hash.

The current unit tests don't actually run this new code because the new code only runs when repoPath is set here

But repoPath isn't set for the tests (because it is added in this PR), so none of this new code is being tested. I think the tests for this would need to be rewritten to work with actual commits and test this code.

I also think maybe we should just not render anything if repoPath is not set, given that the point is to only render links to commits we can verify. This code should probably already be checking metas since it relies on user and repo, but doesn't appear to do (as it does for fullSha1PatternProcessor).

Reply:
> I've added some debugging and it looks like IsCommitExist eventually returns an Object Not Found error for a short hash.

Whoops, did not realize that. I was only testing with full hashes. We may have to switch back to using a `git.NewCommand`.



> There should be some tests added for this

Good point. Am I correct in assuming that this test would need to be part of the integration tests since it requires a valid repository?



> I also think maybe we should just not render anything if repoPath is not set

Sounds like a good idea. The tests would have to be completely rewritten but that just takes time.

> This code should probably already be checking metas since it relies on user and repo

Okay, I'll add some checks.

Reply:
I *think* you can just set repoPath in the unit tests and then point to one of the repos included for integration tests without making this an integration test itself. So for example if you edit the localMetas in that unit test to include:

```
"repoPath": "../../integrations/gitea-repositories-meta/user3/repo3.git/",
```
Then change the test to use an actual commit in that repo like:
```
var sha = "2a47ca4b614a9f5a43abbd5ad851a54a616ffee6"
```
and update expected test results that should be enough to properly test the code

Reply:
> I _think_ you can just set repoPath in the unit tests and then point to one of the repos included for integration tests without making this an integration test itself. So for example if you edit the localMetas in that unit test to include:
> 
> ```
> "repoPath": "../../integrations/gitea-repositories-meta/user3/repo3.git/",
> ```
> 
> Then change the test to use an actual commit in that repo like:
> 
> ```
> var sha = "2a47ca4b614a9f5a43abbd5ad851a54a616ffee6"
> ```
> 
> and update expected test results that should be enough to properly test the code

Took you advice. Thank you.